### ISSUE TITLE
typeof guard not recognised when minified to `typeof X > "u"`

### ISSUE THREAD

**Issue by NGMarmaduke** [yowainwright/es-check#436]
## Requested Update

Recognise the minified form of a `typeof` guard. Since 9.7.0 (#425), es-check detects bare references to newer globals such as `Float16Array`, and skips them when wrapped in a `typeof X === "undefined"` guard. The guard detection (`getTypeofComparison` in `lib/helpers/astDetector.js`) only accepts `==`, `===`, `!=` and `!==`. esbuild's minifier rewrites the guard to `typeof X > "u"` (and `!==` to `< "u"`), so the guarded reference is flagged.

```js
// passes
var a = typeof Float16Array === "undefined" ? undefined : Float16Array;

// fails: "Unsupported features detected: Float16Array"
var a=typeof Float16Array>"u"?void 0:Float16Array;
```

```sh
es-check es13 ./minified.js --checkFeatures
```

## Why Is This Update Needed?

The second line above is exactly what `esbuild --minify` emits for the first. In practice this fires on any production bundle that includes `get-intrinsic`, which guards every intrinsic that way, including `Float16Array`. 9.6.4 passes the same bundle, since bare global references were not checked before #425.

The rewrite is sound because `"undefined"` is the only `typeof` result that sorts after `"u"` (ECMA-262 §13.5.3, typeof operator results), so `typeof X > "u"` is equivalent to `typeof X === "undefined"` and `typeof X < "u"` to `typeof X !== "undefined"`.

## Are There Examples Of This Requested Update Elsewhere?

esbuild applies this rewrite by default when minifying (`typeof x === "undefined"` to `typeof x > "u"`). Terser leaves the comparison as `"undefined" == typeof x`, which es-check already handles.

Suggested fix: in `getTypeofComparison` / `getTypeofComparisonGuardBranch`, also accept `>` and `<` when the other side is the string literal `"u"`.

**Comment by yowainwright** [comment]
Thanks for calling this out @NGMarmaduke! Please do if you would like, or I will fix asap as well. 🙇
